#Blast Freezer Truck Schedule Optimization for Supply Chain Efficiency
##Initial Report

###Environment Setup and Module Importing

imports necessary Python libraries (pandas, seaborn, numpy) and mounts Google Drive for accessing files, indicating the project's reliance on external datasets and Python's data analysis and visualization capabilities.

In [1]:
import pandas as pd
import seaborn as sns
import numpy as np

In [2]:
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


###Custom Module Importing
It involves importing a custom module named brooks from a specified directory in Google Drive, suggesting the use of custom Python functions or classes defined outside the notebook for specialized tasks.

In [3]:
import sys
sys.path.insert(0, "/content/drive/MyDrive/645")

from brooks import *

###Random Number Generation and Simulation
This part initializes a random number generator with a seed and demonstrates generating triangularly distributed random numbers, a step likely preparatory for simulating processes or behaviors in the project.

Setting the random variables and assuming the mean, mode and max as per triangular distribution.
Assumed: min=24hrs, mode=48hrs, max=72hrs

In [4]:
rg = np.random.default_rng(12345)
for i in range(10):
  print(rg.triangular(24,48,72))

40.183049708944836
43.10250264613606
56.72142044715248
52.687966981830286
45.226356312221824
43.58064464980852
50.4884143804424
38.6668940752798
52.583897474128435
63.8120149481675


###Simulation of Freezing Times
The notebook simulates the total freezing time required for pallets in trucks over a range of scheduled trucks. It uses Monte Carlo simulation techniques to estimate the freezing time, crucial for operational planning.


In [5]:
total_freezing_time_per_week = 34560
trucks_scheduled = range(10, 61, 2)  #randomly scheduling trucks
n = 1000 #Number of simulations

for trucks in trucks_scheduled:
    rg = np.random.default_rng(12345) #Taking a seed value of 12345
    total_freezing_time_list= [] #Creating a list to append the final total times
    for i in range(n):
        total_freezing_time = 0
        pallets_in_truck = 24*trucks
        for pallets in range(pallets_in_truck):
            freezing_time = rg.triangular(48, 55, 72) #Assuming the mean and mode values and taking the max as 72
            total_freezing_time += freezing_time
        total_freezing_time_list.append(total_freezing_time)

    lower_bound = np.percentile(total_freezing_time_list, 2.5)
    upper_bound = np.percentile(total_freezing_time_list, 97.5)



###Statistical Analysis
After simulations, it calculates statistical metrics (like the 2.5th and 97.5th percentiles) to understand the distribution of total freezing times. This helps in risk assessment and decision-making regarding the operational capacity.

In [6]:
print("Total freezing time per week: %f" % (sum(total_freezing_time_list)/n))
no_trucks = ((sum(total_freezing_time_list)/n)/288)/24
print("Number of trucks required are:", no_trucks)

fifty = np.percentile(total_freezing_time_list, q=50)
print ("50th percentile for total freezing time per week: %f" % (fifty))
print(f"Total freezing time per week(50%): Lower Bound: {lower_bound}, Upper Bound: {upper_bound}")

if(fifty>total_freezing_time_per_week):
  print("There is at most a 50% chance the processing time does not exceed 34,560 hours")
if lower_bound <= total_freezing_time_per_week and upper_bound >= total_freezing_time_per_week:
  print("There is at most a 5% chance the processing time does not exceed 34,560 hours")

Total freezing time per week: 83997.884431
Number of trucks required are: 12.152471705820084
50th percentile for total freezing time per week: 83996.599826
Total freezing time per week(50%): Lower Bound: 83606.6733700654, Upper Bound: 84373.76673523802
There is at most a 50% chance the processing time does not exceed 34,560 hours


###Results Interpretation
The results from the simulation indicate that the total average freezing time per week is approximately 83,997.88 hours, which suggests that about 12 trucks would be required to handle the workload, based on an average freezing time of 83996.60 hours (50th percentile). The range between the lower bound (83,606.67 hours) and the upper bound (84,373.77 hours) at the 50th percentile provides a confidence interval for the simulation outcomes. The statement about a 50% chance that processing time does not exceed 34,560 hours seems incorrect given the context, likely indicating a misunderstanding, as the actual processed times far exceed this figure.






